libass/x86/cpuid.asm

```diff
@@ -22,6 +22,31 @@
 
 SECTION .text
 
+;------------------------------------------------------------------------------
+; uint32_t has_cpuid( void );
+;------------------------------------------------------------------------------
+
+INIT_XMM
+cglobal has_cpuid, 0, 2, 0
+%if ARCH_X86_64
+    mov eax, 1
+%else
+    pushfd
+    pop r1
+    mov r0, r1
+    xor r0, 0x00200000
+    push r0
+    popfd
+    pushfd
+    pop r0
+    xor r0, r1
+    and r0, 0x00200000 ; non-zero if bit is writable
+    push r1            ; Restore original EFLAGS
+    popfd
+    mov eax, r0d
```

Comment:
This is `mov eax, eax`. I think it's better to use `eax/ecx` directly instead of `r0/r1` to not trigger unwanted side effects like this. Also maybe rewrite it to use only one register? Something like this (not tested);
```asm
pushfd
mov eax, [esp]
xor [esp], dword 0x00200000
popfd
pushfd
xor eax, [esp]
xor [esp], eax
popfd
and eax, 0x00200000
ret
```

Reply:
What side-effects does `mov eax, eax` have? Afaik this is just a no-op.

Regarding `r0/r1`, sure I can change it to use `eax/ecx`, if altering `ecx` is allowed in all supported calling conventions and you think avoiding one potential no-op is worth it.

As for using only one register, I don't see what benefit this is supposed to bring. Even 32bit x86 has 3 caller saved registers in the calling convention I know about. Are compilers (with LTO?) capable of omitting saving some normally caller-saved registers when they know the called function doesn't use them? And if so, does this really matter here; `has_cpuid` is not performance-critical and iinm only called once per `ASS_Renderer`.

Reply:
> Are compilers (with LTO?) capable of omitting saving some normally caller-saved registers when they know the called function doesn't use them?

They are, but here the C compiler won't know that the nasm-compiled function doesn't use some registers.

Reply:
Now using `eax`, `ecx` and `cglobal 0,0,0` instead of `r0/r1`.